File: frontend/src/constants/noteMethods.ts
import type { NotebookMethod } from '../types';

/**
 * Display metadata for each note-taking method.
 * Used in the notebook creation flow and anywhere methods are listed in the UI.
 */
export const METHODS: {
  id: NotebookMethod;
  icon: string;
  name: string;
  description: string;
}[] = [
  {
    id: 'cornell',
    icon: '📝',
    name: 'Cornell',
    description: 'Organized sections for cues, notes, and summary',
  },
  {
    id: 'sentence',
    icon: '📋',
    name: 'Sentence',
    description: 'Each thought as a numbered line',
  },
  {
    id: 'outlining',
    icon: '📑',
    name: 'Outlining',
    description: 'Hierarchical structure with topics and subtopics',
  },
  {
    id: 'charting',
    icon: '📊',
    name: 'Charting',
    description: 'Compare and categorize with structured sections',
  },
  {
    id: 'blank',
    icon: '📄',
    name: 'Blank',
    description: 'Free-form notes with no structure',
  },
];

File: tablet/lib/constants/note_methods.dart
import 'package:flutter/material.dart';

/// Metadata for a note-taking method.
class NoteMethod {
  /// API identifier (matches backend enum).
  final String id;

  /// Material icon for display.
  final IconData icon;

  /// Human-readable name.
  final String name;

  /// Short description.
  final String description;

  const NoteMethod({
    required this.id,
    required this.icon,
    required this.name,
    required this.description,
  });
}

/// All available note-taking methods.
///
/// Order matches the frontend's `noteMethods.ts`.
const List<NoteMethod> noteMethods = [
  NoteMethod(
    id: 'cornell',
    icon: Icons.edit_note,
    name: 'Cornell',
    description:
        'Organized sections for cues, '
        'notes, and summary',
  ),
  NoteMethod(
    id: 'sentence',
    icon: Icons.format_list_numbered,
    name: 'Sentence',
    description:
        'Each thought as a numbered line',
  ),
  NoteMethod(
    id: 'outlining',
    icon: Icons.segment,
    name: 'Outlining',
    description:
        'Hierarchical structure with '
        'topics and subtopics',
  ),
  NoteMethod(
    id: 'charting',
    icon: Icons.table_chart_outlined,
    name: 'Charting',
    description:
        'Compare and categorize with '
        'structured sections',
  ),
  NoteMethod(
    id: 'blank',
    icon: Icons.description_outlined,
    name: 'Blank',
    description:
        'Free-form notes with no structure',
  ),
];

/// Returns the [NoteMethod] matching [id],
/// or the 'blank' method as fallback.
NoteMethod noteMethodById(String id) {
  return noteMethods.firstWhere(
    (m) => m.id == id,
    orElse: () => noteMethods.last,
  );
}
